src/inet/UDPEndPointImplSockets.cpp

```diff
@@ -274,6 +274,9 @@ CHIP_ERROR UDPEndPointImplSockets::ListenImpl()
 
 CHIP_ERROR UDPEndPointImplSockets::SendMsgImpl(const IPPacketInfo * aPktInfo, System::PacketBufferHandle && msg)
 {
+    // Ensure packet buffer is not null
+    VerifyOrReturnError(!msg.IsNull(), CHIP_ERROR_NO_MEMORY);
```

Comment:
A sanity check here is probably ok, but why is someone passing in null?  Is this the `CloneData` calls in `src/lib/dnssd/minimal_mdns/Server.cpp`?  If so, _those_ should be checking that the clone succeeded...

Reply:
That's correct, @bzbarsky-apple . It's CloneData which is why it is possible to see the msg being null.

So you'd have a preference of having the check inside `UDPEndPoint::SendMsg()`?

Reply:
I'd prefer that whoever does the `CloneData` checks the result before passing it around.  They are the ones who are in the best position to decide what to do if the clone fails.

Reply:
That's a fair point. So we can keep this check as sanity. The caller is from the ServerBase::BroadcastImpl() inside Server.cpp. Will get another commit up with said recommendations.

Reply:
I've added the checks into the MinimalMdns server in the latest commit, please feel free to review.